llarp/router_contact_remote.cpp

```diff
@@ -0,0 +1,69 @@
+#include "constants/version.hpp"
+#include "crypto/crypto.hpp"
+#include "net/net.hpp"
+#include "router_contact.hpp"
+#include "util/bencode.hpp"
+#include "util/buffer.hpp"
+#include "util/file.hpp"
+#include "util/time.hpp"
+
+#include <oxenc/bt_serialize.h>
+
+namespace llarp
+{
+  RemoteRC::RemoteRC(std::string payload)
+  {
+    try
+    {
+      oxenc::bt_dict_consumer btdc{payload};
+      bt_load(btdc);
+      bt_verify(btdc);
+    }
+    catch (const std::exception& e)
+    {
+      log::warning(logcat, "Failed to parse RemoteRC: {}", e.what());
+      throw;
+    }
+  }
+
+  void
+  RemoteRC::bt_encode(oxenc::bt_dict_producer& btdp) const
+  {
+    (void)btdp;
+
+    // TODO: implement append_encoded in oxenc so we can finish this implementation.
+    // It is almost identical to the implementation of LocalRC::bt_encode, except the
+    // call to append_signature is append_encoded.
```

Comment:
As described this won't work, because it will potentially break signatures (if the RC we received had extra keys we didn't understand, from a future Lokinet).  We'll have to instead get the entire, untouched, encoded RC as we received it and push that out.

Reply:
For example, future lokinet might decide it doesn't care about ipv6 anymore and stops parsing the ipv6 field entirely, so then this:
```json
{"4": "abcdpp", "6": "whateverpp", "p": "...pubkey...", "~": "xyz..."}
```
would get round-trip re-encoded as:
```json
{"4": "abcdpp", "p": "...pubkey...", "~": "xyz..."}
```
but now the `xyz...` won't match anymore because the "6" will be missing from the serialized dict output.

Reply:
I know it wont work! I remember your explanation on that. To clarify, I meant we need `append_encoded` for adding the signature to the end, not the individual key/value pairs. I can explain more over mumble when you're online